### ISSUE TITLE
Preferred way to do routing

### ISSUE THREAD

**Issue by mklueh** [daaru00/gridsome-plugin-i18n#1]
First of all, very promising plugin, thank you for that. I'm just wondering how your preferred way of doing routing within the same language looks like?

**Comment by daaru00** [comment]
Hi @mklueh,
thank you! 

> I'm just wondering how your preferred way of doing routing within the same language looks like?

I currently doing a test valorizing base meta HTML tag with site base URL with current language path. All page's links must be relative in order to make this work.

Maybe a better could be directly integrate the logic inside Vue Router to let using [g-link directive](https://github.com/gridsome/gridsome/blob/4d2271222cb44e94b9e6f7c0ca04740226c0a7ce/gridsome/app/directives/link.js#L32) without any interference. Just a `beforeEach` that check presence of locale path, if does not exist the middleware add it.

Similar to [Nuxt approach](https://github.com/nuxt-community/nuxt-i18n/blob/master/src/plugins/routing.js#L58) 
```javascript
    const isPrefixed = (
      // don't prefix default locale
      !(locale === defaultLocale && strategy === STRATEGIES.PREFIX_EXCEPT_DEFAULT) &&
        // no prefix for any language
        !(strategy === STRATEGIES.NO_PREFIX) &&
        // no prefix for different domains
        !i18n.differentDomains
    )

    const path = (isPrefixed ? `/${locale}${route.path}` : route.path)

    localizedRoute.path = path
```
don't know if this can be also used in Gridsome or need a different solution.

**Comment by mklueh** [comment]
@daaru00 sounds promising. Are you able to implement that into Gridsome? Otherwise you could create an issue there and link to your Plugin. I think a lot of people are looking forward to that

**Comment by daaru00** [comment]
Hi @mklueh,

I found a more elegant way to rewrite path using [beforeResolve](https://github.com/daaru00/gridsome-plugin-i18n/pull/2/files#diff-24bdcc0ccc3743ecc83a0bdcc0ba0e2dR64) router hook. This should keep a full compatibility with `g-link` directive.

I also added a `$tp` helper function to cover multiple cases and generate page's html with correct localized links (more SEO friendly).

I just opened a PR #2 with this improvements, can you take a look?

**Comment by mklueh** [comment]
Hi @daaru00,

I´ve looked into it and as far as I can judge it it looks amazing. However, I have one question regarding this line https://github.com/daaru00/gridsome-plugin-i18n/blob/61d604d8d4de2e2d80d879120f7571177eb35ac3/gridsome.client.js#L70

You are checking if it runs on the server, but does it ever run on the client when the Gridsome application is built? I had the impression that the router get´s resolved (in contrast to dev mode) and won´t exist in the client, as the finished application will not be a single page application but a static site consisting of multiple subpages.

In which cases is the condition "not" server?

**Comment by daaru00** [comment]
Hi @mklueh 

all the Vue client application is bootstrapped during build phase. Gridsome use Vue SSR to do that and router is bootstrap too ([here Vue SSR doc about it](https://ssr.vuejs.org/guide/routing.html#routing-with-vue-router)) and so is fully active, even `beforeResolve` hook where I run the rewrite logic is also executed during build phase. In that moment `process.isServer` is true because is executed by NodeJS (inside npm/yarn) that run it and not the browser.

I excluded the path rewrite during this phase because of an error raised by Gridsome. Is fully justified because when it start building `/test` my hooked logic rewrite it to `/en/test` and this break all build process because page's path is a different then what Gridsome expect (created by `createPage` API).

have I clarified your doubt?

**Comment by daaru00** [comment]
Hi @mklueh,

just released version 1.2.0, you can update this module:
```bash
npm update gridsome-plugin-i18n
```
and routing rewrite system should start works.
Let me know if works as expected.

**Comment by mklueh** [comment]
Hi @daaru00,

thanks for the explanation. I really like the API of your plugin, very clear and configurable.
I´ve now had time to try it out and I´m running into this issue:

```
Create GraphQL schema - 0.1s
Error: child "path" fails because ["path" with value "\de-de\404\" fails to match the leading slash pattern]
    at validate (C:\Users\MyPC\IdeaProjects\gridsome-i18n\node_modules\gridsome\lib\pages\schemas.js:54:11)
    at Pages.createPage (C:\Users\MyPC\IdeaProjects\gridsome-i18n\node_modules\gridsome\lib\pages\pages.js:165:21)
    at createPage (C:\Users\MyPC\IdeaProjects\gridsome-i18n\node_modules\gridsome\lib\app\actions.js:260:24)
    at VueI18n.createManagedPages (C:\Users\MyPC\IdeaProjects\gridsome-i18n\node_modules\gridsome-plugin-i18n\gridsome.server.js:47:9)
    at Plugins.run (C:\Users\MyPC\IdeaProjects\gridsome-i18n\node_modules\gridsome\lib\app\Plugins.js:141:17)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Plugins.createPages (C:\Users\MyPC\IdeaProjects\gridsome-i18n\node_modules\gridsome\lib\app\Plugins.js:116:5)
```

with this configuration


 ```
{
            use: "gridsome-plugin-i18n",
            options: {
                locales: [ 
                    'de-de',
                    'en-us'
                ],
                pathAliases: {
                    'de': 'de',
                    'en': 'en'
                },
                fallbackLocale: 'en-us',
                defaultLocale: 'en-us', 
                messages: {
                    'de-de': require('./src/locales/de-de.json'),
                    'en-gb': require('./src/locales/en-us.json'),
                }
            }
        },
```

Can you reproduce that on your side?
